examples/nextjs-with-typescript/pages/MediaChromeReactSimple.tsx

```diff
@@ -0,0 +1,343 @@
+    <button
+      style={{ cursor: 'pointer' }}
+      disabled={!mediaSubtitlesList?.length}
+      onClick={() => {
+        const type = showingSubtitles
+          ? MediaUIEvents.MEDIA_DISABLE_SUBTITLES_REQUEST
+          : MediaUIEvents.MEDIA_SHOW_SUBTITLES_REQUEST;
+        const detail = showingSubtitles
+          ? mediaSubtitlesShowing
+          : [mediaSubtitlesList[0]];
+        dispatch({ type, detail });
+      }}
+    >
+      {showingSubtitles ? 'Disable Captions' : 'Enable Captions'} React Only!
+    </button>
+  );
+};
+
+const PipButton = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaIsPip = useSelector((state) => state.mediaIsPip);
+  return (
+    <button
+      style={{ cursor: 'pointer' }}
+      onClick={() => {
+        const type = mediaIsPip
+          ? MediaUIEvents.MEDIA_EXIT_PIP_REQUEST
+          : MediaUIEvents.MEDIA_ENTER_PIP_REQUEST;
+        dispatch({ type });
+      }}
+    >
+      {!mediaIsPip ? 'Enter Pip' : 'Exit Pip'} React Only!
+    </button>
+  );
+};
+
+const FullscreenButton = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaIsFullscreen = useSelector((state) => state.mediaIsFullscreen);
+  return (
+    <button
+      style={{ cursor: 'pointer' }}
+      onClick={() => {
+        const type = mediaIsFullscreen
+          ? MediaUIEvents.MEDIA_EXIT_FULLSCREEN_REQUEST
+          : MediaUIEvents.MEDIA_ENTER_FULLSCREEN_REQUEST;
+        dispatch({ type });
+      }}
+    >
+      {!mediaIsFullscreen ? 'Enter Fullscreen' : 'Exit Fullscreen'} React Only!
+    </button>
+  );
+};
+
+const TimeRange = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaCurrentTime = useSelector((state) => state.mediaCurrentTime);
+  // @ts-ignore
+  const mediaDuration = useSelector((state) => state.mediaDuration);
+  return (
+    <div>
+      <span>Time Range</span>
+      <input
+        type="range"
+        min={0}
+        max={Number.isNaN(mediaDuration) ? 0 : mediaDuration}
+        value={mediaCurrentTime}
+        step={0.1}
+        onChange={(event) => {
+          const type = MediaUIEvents.MEDIA_SEEK_REQUEST;
+          const detail = +event.target.value;
+          dispatch({ type, detail });
+        }}
+      />
+    </div>
+  );
+};
+
+const VolumeRange = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaVolume = useSelector((state) => state.mediaVolume);
+  return (
+    <div>
+      <span>Volume Range</span>
+      <input
+        type="range"
+        min={0}
+        max={1}
+        value={mediaVolume}
+        step={0.1}
+        onChange={(event) => {
+          const type = MediaUIEvents.MEDIA_VOLUME_REQUEST;
+          const detail = +event.target.value;
+          dispatch({ type, detail });
+        }}
+      />
+    </div>
+  );
+};
+
+const Video = ({ src }: { src?: string }) => {
+  const mediaRefCallback = useMediaRefCallback();
+  return (
+    <video
+      ref={mediaRefCallback}
```

Comment:
I'm interested to understand better how this ultimately connects to the store.

Reply:
The tl;dr - it's basically just a specialized version of the examples that use `dispatch()`. Happy to discuss further. Here are some relevant tidbits:
1. `ref` callback (React component feature) - https://react.dev/reference/react-dom/components/common#ref-callback
2. `useMediaRefCallback()` definition - https://github.com/muxinc/media-chrome/pull/549/files#diff-72149d92e0737edb63a3cc1ac7aa148e6cf8b9105cb2d07eb7bd8dc05728d8bfR55-R60 (creates a function that will cause a `dispatch()` of the `"mediaelementchangerequest"` "action")
3. What media store does when it receives a `"mediaelementchangerequest"` "action" - https://github.com/muxinc/media-chrome/pull/549/files#diff-a13fb0284c9e2463ff5bd20405c5a92ac40b6d35dea7320d5e34bb3085591bf0R1000-R1002 (figures out what the state should be based on the newly added/replaced media "state owner", including potentially tearing down things like event listeners)